pkg/cmd/pulumi/whoami.go

```diff
@@ -58,20 +59,50 @@ func newWhoAmICmd() *cobra.Command {
 			}
 
 			if verbose {
+				if jsonOut {
+					return outputWhoAmIJSON(WhoAmIJSON{
+						User:          name,
+						Organizations: orgs,
+						BackendURL:    b.URL(),
+					})
+				}
 				fmt.Printf("User: %s\n", name)
 				fmt.Printf("Organizations: %s\n", strings.Join(orgs, ", "))
 				fmt.Printf("Backend URL: %s\n", b.URL())
 			} else {
+				if jsonOut {
+					return outputWhoAmIJSON(WhoAmIJSON{
+						User: name,
+					})
+				}
 				fmt.Println(name)
 			}
 
 			return nil
 		}),
 	}
 
+	cmd.PersistentFlags().BoolVarP(
+		&jsonOut, "json", "j", false, "Emit output as JSON")
+
 	cmd.PersistentFlags().BoolVarP(
 		&verbose, "verbose", "v", false,
 		"Print detailed whoami information")
 
 	return cmd
 }
+
+// WhoAmIJSON is the shape of the --json output of this command.
+type WhoAmIJSON struct {
+	User          string   `json:"user"`
+	Organizations []string `json:"organizations,omitempty"`
+	BackendURL    string   `json:"backendUrl"`
+}
+
+func outputWhoAmIJSON(output WhoAmIJSON) error {
```

Comment:
smallest of nits:
We don't need this function if we do the `if jsonOut` thing I suggested above.
Would you mind inlining this function up there?

```
if json {
  return printJSON(WhoAmIJSON{...})
}
```

Reply:
You're right. SGTM!